Shared/sdk/SharedUtil.Misc.h

```diff
@@ -604,24 +546,23 @@ namespace SharedUtil
     //
     // Note: IDs run from 1 to Capacity
     //
-    template <typename T, unsigned long INITIAL_MAX_STACK_SIZE>
+    template <typename T, std::uint32_t INITIAL_MAX_STACK_SIZE>
     class CStack
     {
     public:
-        CStack()
+        CStack() noexcept
         {
             m_ulCapacity = 0;
             ExpandBy(INITIAL_MAX_STACK_SIZE - 1);
         }
 
-        unsigned long GetCapacity() const { return m_ulCapacity; }
-
-        unsigned long GetUnusedAmount() const { return m_Queue.size(); }
+        std::uint32_t GetCapacity() const noexcept { return m_ulCapacity; }
+        std::uint32_t GetUnusedAmount() const noexcept { return m_Queue.size(); }
 
-        void ExpandBy(unsigned long ulAmount)
+        void ExpandBy(std::uint32_t amount)
```

Comment:
size_t

Reply:
The same for all quantity types above and below.

Reply:
@TracerDS